### PR TITLE
Fix #2: Generalize the script to work on any repository.

### LINKED ISSUE #2: Make this a general script to be used by any other project.
Looked into doing something similar for [Focus](https://github.com/mozilla-mobile/focus-android/) that needs to verify the beta versions similarly to know when to uplift strings ([PR adding the workflow for syncing strings](https://github.com/mozilla-mobile/focus-android/pull/7124)) but it seems like we can adapt this project to use `GITHUB_REPOSITORY` instead of 

https://github.com/mozilla-mobile/fenix-beta-version/blob/0bab3e17e9047f00a5f1e00a880be76045fa70b8/src/fenix-beta-version.py#L61-L69

to avoid having to duplicate this for every other Mozilla project that needs this functionality.

### PR DESCRIPTION
**Pull request by Mugurell**
This replaces the <pre>os.getenv("GITHUB_REPOSITORY_OWNER")/<b>fenix</b></pre> hardcode with <pre>os.getenv("GITHUB_REPOSITORY")</pre> and in that allow the script to extract the last major version of a beta release from any repository.

The refactoring included many renamings to avoid specifying "fenix".
Repository [can be renamed](https://docs.github.com/en/repositories/creating-and-managing-repositories/renaming-a-repository) also to complete the refactoring but this would probably be a breaking change for users of this workflow.

This introduces a breaking change (which can be avoided by creating a new tag for the project, maybe `3.0.0`): the output was also made generic by now being named `beta_version`.

The result of this workflow after refactoring was tested on https://github.com/Mugurell/TestGithubActions/runs/6655281873

### CODE REVIEW

File: src/extract-major-beta-version.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python3
```
Comment (escapewindow) [inline src/extract-major-beta-version.py:1]: (This would have been easier to review with a `git mv`)

File: src/extract-major-beta-version.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return int(matches[1])
+    raise Exception(f"Unexpected release branch name: {branch_name}")
+
+
+def get_release_branches(repo):
+    return [branch.name for branch in repo.get_branches()
+            if re.match(r"^releases[_/]v\d+\.0\.0$", branch.name)]
+
+
+def get_latest_beta_major_version(repo):
```
Comment (escapewindow) [inline src/extract-major-beta-version.py:29]: Does this get the latest major version number, regardless of whether it's `beta` ?

File: src/extract-major-beta-version.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+from github import Github, InputGitAuthor, enable_console_debug_logging
+
+
+#
+# All these are lifted from relbot/utils.py
+#
+
+
+def major_version_from_release_branch_name(branch_name):
+    if matches := re.match(r"^releases[_/]v(\d+)\.0\.0$", branch_name):
```
Comment (escapewindow) [inline src/extract-major-beta-version.py:19]: We're using this regex in a number of places. Using the `releases/` format is no longer supported; we could update these to just support `releases_`.

**Review by escapewindow — APPROVED** [review]
I think this works, thank you!
I don't think my comments are blocking, but if you agree, we can wait to merge until you make those changes.
When we're ready, I can merge and tag.

**Comment by escapewindow** [comment]
I'll go ahead and land. We can land any changes as followups.

File: src/extract-major-beta-version.py (hunk 1 above)
Comment (Mugurell) [reply src/extract-major-beta-version.py:1]: Tried to keep the file history and checking now the just merged files it seems blame does differentiate the changes - https://github.com/mozilla-mobile/fenix-beta-version/blame/main/src/extract-major-beta-version.py.
Indeed if looking over the files in this PR - https://github.com/mozilla-mobile/fenix-beta-version/pull/3/files the `extract-major-beta-version.py` appears as a new file (all new content) but by having the renaming included in a separate commit only the actual changes - https://github.com/mozilla-mobile/fenix-beta-version/pull/3/commits/da0124289c71c6006e35e48b9a9e2f06951f4a8c seem to have gotten merged.

File: src/extract-major-beta-version.py (hunk 2 above)
Comment (Mugurell) [reply src/extract-major-beta-version.py:29]: Indeed. 
Will update, thank you!

File: src/extract-major-beta-version.py (hunk 3 above)
Comment (Mugurell) [reply src/extract-major-beta-version.py:19]: Sure! Thank you!

**Comment by Mugurell** [comment]
> I'll go ahead and land. We can land any changes as followups.

Thank you!
Opened https://github.com/mozilla-mobile/fenix-beta-version/pull/5 to address your suggestions.

**Pull request merged**